File: src/main/java/org/dementhium/content/clans/ClanPacket.java
package org.dementhium.content.clans;

import org.dementhium.model.player.Player;
import org.dementhium.net.message.MessageBuilder;
import org.dementhium.net.message.Message.PacketType;
import org.dementhium.util.Misc;


/**
 * @author 'Mystic Flow
 */
public class ClanPacket {
	public static void sendClanList(Player p, Clan clan) {
		MessageBuilder bldr = new MessageBuilder(23, PacketType.VAR_SHORT);
		if (clan != null) {
			bldr.writeRS2String(Misc.formatPlayerNameForDisplay(clan.getOwner()));
			bldr.writeByte(0);
			bldr.writeLong(Misc.stringToLong(clan.getName()));
			bldr.writeByte(clan.getKickReq());
			bldr.writeByte(clan.getMembers().size());
			for (Player pl : clan.getMembers()) {
				bldr.writeRS2String(Misc.formatPlayerNameForDisplay(pl.getUsername()));
				bldr.writeByte(p.getConnection().isInLobby() ? 1 : 0); // Need to figure this out
				if(p.getConnection().isInLobby()) {
					bldr.writeRS2String(pl.getConnection().isInLobby() ? "Lobby" : "ExemptionX");
				}
				bldr.writeShort(2); // Status
				bldr.writeByte(clan.getRank(pl));
				bldr.writeRS2String(pl.getConnection().isInLobby() ? "Lobby" : "<col=00ff00>ExemptionX");
			}
		}
		p.getConnection().write(bldr.toMessage());
	}
}

File: src/main/java/org/dementhium/content/clans/ClanPacket.kt
package org.dementhium.content.clans

import org.dementhium.model.player.Player
import org.dementhium.net.message.Message.PacketType
import org.dementhium.net.message.MessageBuilder
import org.dementhium.util.Misc

/**
 * @author 'Mystic Flow
 */
object ClanPacket {
    fun sendClanList(toPlayer: Player, clan: Clan) = with(MessageBuilder(23, PacketType.VAR_SHORT)) {
        clan.whenValidClan {
            writeRS2String(Misc.formatPlayerNameForDisplay(clan.owner))
            writeByte(0)
            writeLong(Misc.stringToLong(clan.name))
            writeByte(clan.kickReq)
            writeByte(clan.getMembers().size)
            for (clanMember in clan.getMembers()) {
                writeRS2String(Misc.formatPlayerNameForDisplay(clanMember.username))
                writeByte(if (toPlayer.connection.isInLobby) 1 else 0) // Need to figure this out
                if (toPlayer.connection.isInLobby) {
                    writeRS2String(if (clanMember.connection.isInLobby) "Lobby" else "ExemptionX")
                }
                writeShort(2) // Status
                writeByte(clan.getRank(clanMember))
                writeRS2String(if (clanMember.connection.isInLobby) "Lobby" else "<col=00ff00>ExemptionX")
            }
        }
        toPlayer.connection.write(toMessage())
    }
}
